src/tests/utils/renderUtils.tsx

```diff
@@ -0,0 +1,21 @@
+import React, { PropsWithChildren } from "react";
+import { render } from "@testing-library/react";
+import type { RenderOptions } from "@testing-library/react";
+import type { PreloadedState } from "@reduxjs/toolkit";
+import { Provider } from "react-redux";
+import { AppStore, RootState, setupStore } from "../../app/store";
+
+interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
+  preloadedState?: PreloadedState<RootState>;
+  store?: AppStore;
+}
+
+export function renderWithProviders(
```

Comment:
This is taken straight from the redux testing docs. See https://redux.js.org/usage/writing-tests#setting-up-a-reusable-test-render-function